ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,290 @@
+    """Cloud Trace span exporter for OpenTelemetry.
+
+    Args:
+        project_id: ID of the cloud project that will receive the traces.
+        client: Cloud Trace client. If not given, will be taken from gcloud
+            default credentials
+    """
+
+    def __init__(
+        self, project_id=None, client=None,
+    ):
+        self.client = client or TraceServiceClient()
+        if not project_id:
+            _, self.project_id = google.auth.default()
+        else:
+            self.project_id = project_id
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        See: https://cloud.google.com/trace/docs/reference/v2/rest/v2/projects.traces/batchWrite
+
+        Args:
+            spans: Tuple of spans to export
+        """
+        cloud_trace_spans = []
+        for span in self.translate_to_cloud_trace(spans):
+            try:
+                cloud_trace_spans.append(self.client.create_span(**span))
+            # pylint: disable=broad-except
+            except Exception as ex:
+                logger.warning("Error %s when creating span %s", ex, span)
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_cloud_trace(
+        self, spans: Sequence[Span]
+    ) -> List[Dict[str, Any]]:
+        """Translate the spans to Cloud Trace format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        cloud_trace_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = _get_hexidecimal_trace_id(ctx.trace_id)
+            span_id = _get_hexidecimal_span_id(ctx.span_id)
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if span.parent:
+                parent_id = _get_hexidecimal_span_id(span.parent.span_id)
+
+            start_time = _get_time_from_ns(span.start_time)
+            end_time = _get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+
+            cloud_trace_spans.append(
+                {
+                    "name": span_name,
+                    "span_id": span_id,
+                    "display_name": _get_truncatable_str_object(span.name),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
```

Comment:
One note about these: right now there's an issue with DefaultSpans being created rather than an sdk.Span, because that's the way a sample omits a lot of expensive processing.

But this isn't the first time I've seen this caveat, and the big hoops that we have to jump to try to support both DefaultSpan and Span. 

Unfortunately today you will have to consider the scenario of a DefaultSpan being passed in instead, so these should be changed to hasattr() guarded checks.

Reply:
Per our offline discussion, this issue will be resolved in the layer above the SpanProcessor in a different PR